scripts/github_api

```diff
@@ -0,0 +1,118 @@
+#!/bin/python2
+from functools import partial
+import argparse
+import json
+import re
+import requests
+import sys
+
+CLIENT_ERROR = 422
+API_BASE_URL = "https://api.github.com/"
+
+
+def request(*args, **kwargs):
+    raise Exception("The '_setup_auth' function must be called once "
+                    "before a request can be made")
+
+
+def _setup_auth(user, password):
+    auth = (user, password)
+    global request
+    request = partial(_request, auth)
+
+
+def _request(auth, method, endpoint, **kwargs):
+    methods = {
+        'POST': requests.post
+    }
+
+    r = methods[method](endpoint, auth=auth, **kwargs)
+    response = json.loads(r.text)
+
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        if r.status_code in [CLIENT_ERROR]:
+            error_msg = response['errors'][0]['message']
+            raise Exception(error_msg)
+        raise
+
+    return response
+
+
+class PullRequest:
+    """
+    https://developer.github.com/v3/pulls/
+    """
+    def __init__(self, repo):
+        self.repo = repo
+        self._endpoint = API_BASE_URL + "repos/%(repo)s/pulls"
+
+    def open(self, title, body, src_branch, dst_branch):
+        endpoint = self._endpoint % dict(repo=self.repo)
```

Comment:
minor: I'd prefer to avoid HTTP library specific terminology, such as "endpoint" and "payload" and use the names defined by the HTTP protocol: URL and body.